### PR TITLE
[DRAFT] Added basic byte array support

### PR DESCRIPTION
**Pull request by Dacarnix**

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -25,7 +25,7 @@ android {
     compileSdkVersion 28
 
     defaultConfig {
-        minSdkVersion 16
```
Comment (erickzanardo) [inline android/build.gradle]: Don't forget to add a disclaimer on README saying that we only support Android 6.0

File: android/src/main/java/xyz/luan/audioplayers/ByteDataSource.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import android.media.MediaDataSource;
+import android.os.Build;
+
+import java.io.*;
+
+public class ByteDataSource extends MediaDataSource {
+    private byte[] data;
+
+    public ByteDataSource(byte[] realData)
```
Comment (erickzanardo) [inline android/src/main/java/xyz/luan/audioplayers/ByteDataSource.java]: put the { on the same line of the method declaration

File: android/src/main/java/xyz/luan/audioplayers/ByteDataSource.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    } 
+
+    @Override
+    public synchronized void close() throws IOException {
+        data = null;
+    }
+
+    @Override
+    public synchronized int readAt(long position, byte[] buffer, int offset, int size) {
+        if (position >= data.length)
```
Comment (erickzanardo) [inline android/src/main/java/xyz/luan/audioplayers/ByteDataSource.java]: put the { on the same line of the if

File: android/src/main/java/xyz/luan/audioplayers/ByteDataSource.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            return -1;
+        }
+
+        if (position + size > data.length)
```
Comment (erickzanardo) [inline android/src/main/java/xyz/luan/audioplayers/ByteDataSource.java]: here too

File: ios/Classes/AudioplayersPlugin.m [hunk 5]
```diff
@@ -29,11 +29,6 @@ @implementation AudioplayersPlugin {
 FlutterMethodChannel *_channel_audioplayer;
 bool _isDealloc = false;
 
-FlutterEngine *_headlessEngine;
```
Comment (erickzanardo) [inline ios/Classes/AudioplayersPlugin.m]: Any reason for that amount of code removal from this file

File: android/build.gradle (hunk 1 above)
Comment (Dacarnix) [reply android/build.gradle]: Wow thanks for the quick comments. I really wasn't expecting such a fast review :D 

I had this code sitting around in a branch for a month or two, and decided to put it out as a draft so I didn't forget about it / motivate me to finish it. I'll do a huge clean up tomorrow or so :+1:

**Pull request closed without merging**